### PR TITLE
[Nit] Merkle Tree Update

### PR DESCRIPTION
**Pull request by d0cd**
This PR,
- fixes the `*update_and_update_many_match` tests.
- improves documentation for the `update_many` algorithm.

### CODE REVIEW

**Review by ljedrz — APPROVED** [review]
Looks fine to me, though I'm surprised that even though the first commit suggests that it fixes the test(s), there was no failure before.

**Comment by d0cd** [comment]
The test was changed #1501.
Originally, this test ran `update` on a set of non-unique updates, ran `update_many` on the subset of unique updates and checked that the roots match.
Currently, the test runs `update` and `update_many` on the unique updates and checks that the roots match.
This PR reverts back to the original tests

**Review by raychu86 — APPROVED** [review]

File: console/collections/src/merkle_tree/mod.rs [hunk 1]
```diff
@@ -370,6 +372,10 @@ impl<E: Environment, LH: LeafHash<Hash = PH::Hash>, PH: PathHash<Hash = Field<E>
                     false => false,
                 };
                 // Get the sibling hash.
+                // Note: This algorithm assumes that the sibling hash is either the next hash in the vector,
+                // or in the original Merkle tree. Consequently, updates need to be provided in sequential order.
+                // This is enforced by the type of `updates: `BTreeMap<usize, LH::Leaf`.
```
Comment (howardwu) [inline console/collections/src/merkle_tree/mod.rs]:
```suggestion
                // This is enforced by the type of `updates: `BTreeMap<usize, LH::Leaf>`.
```

**Review by howardwu — APPROVED** [review]
LGTM

**Pull request merged**